packages/client/src/connection.ts

```diff
@@ -75,18 +85,21 @@ export type ConnectionOptionsWithDefaults = Required<Omit<ConnectionOptions, 'tl
 
 export const LOCAL_TARGET = '127.0.0.1:7233';
 
-export function defaultConnectionOpts(): ConnectionOptionsWithDefaults {
+function addDefaults(options: ConnectionOptions): ConnectionOptionsWithDefaults {
+  const { channelArgs, interceptors, ...rest } = options;
   return {
     address: LOCAL_TARGET,
     credentials: grpc.credentials.createInsecure(),
     channelArgs: {
       'grpc.keepalive_permit_without_calls': 1,
       'grpc.keepalive_time_ms': 30_000,
       'grpc.keepalive_timeout_ms': 15_000,
+      ...channelArgs,
     },
-    interceptors: [makeGrpcRetryInterceptor(defaultGrpcRetryOptions())],
+    interceptors: interceptors ?? [makeGrpcRetryInterceptor(defaultGrpcRetryOptions())],
```

Comment:
So if users add interceptors, they are expected to themselves add the `grpc retry interceptor`, with proper options, to there interceptors array? Have we documented this anywhere?

Reply:
That's by design and documented in `ConnectionOptions.interceptors`. Added this to help:

https://github.com/temporalio/sdk-typescript/pull/908